tests/test_docs/test_build.py

```diff
@@ -0,0 +1,279 @@
+import json
+import logging
+from pathlib import Path
+from textwrap import dedent
+
+import pydantic
+import pytest
+
+from docs.build import (
+    build_schemas_from_source,
+    dump_json_schema,
+    get_schema_from_module,
+    load_schema_module,
+)
+
+
+def test_load_schema_module(tmp_path):
```

Comment:
Shouldn't our tests target only the `run` method? Other methods here are very much "private" and `run` is the only one to be called by anything external. One reason is to allow for future refactoring: as it is right now, if we change/remove one of those methods but the overall behaviour is the same, tests would crash. We only want tests to crash if the behaviour changed, not if the implementation changed.

Reply:
I think it's standard practice to unit test the smallest possible testable components (which in this case is functions). If someone renames the functions being tested, they'll have to update the tests.

Reply:
Now, we could consider all those methods to be public and isolated and should be called units. The problem that persists is that no test validates what we actually care about.

Reply:
Besides, these methods are less private than you think. Grabbing the schema from the module is the same function that the test runner will use to grab the schema to validate the databags. Now that I inspect the tests a bit better that stands out.

Reply:
removed the offending methods and put them in the interface tester pytest plugin